esp-hal/src/spi/master.rs

```diff
@@ -3576,8 +3578,13 @@ struct SpiFuture<'a> {
 }
 
 impl<'a> SpiFuture<'a> {
-    pub fn new(driver: &'a Driver) -> Self {
-        Self { driver }
+    fn setup(driver: &'a Driver) -> impl Future<Output = Self> {
+        core::future::poll_fn(move |cx| {
```

Comment:
This never returns pending, is the poll_fn actually doing anything?

Reply:
it lets me access the context for the waker registration.

Reply:
While there is already a comment in `execute_operation_async` maybe a comment here might help to prevent "future us" from accidentally reverting this?